src/main/java/pers/gwyog/gtneioreplugin/plugin/gregtech5/PluginGT5Base.java

```diff
@@ -54,25 +45,51 @@ protected static String getGTOreUnlocalizedName(short index) {
-            if (GuiDraw.getMousePosition().y > (int) (Minecraft.getMinecraft().currentScreen.height * 0.6f) && GuiDraw.getMousePosition().y < (int) (Minecraft.getMinecraft().currentScreen.height * 0.8f)) {
-                List<String> dims = DimensionHelper.convertCondensedStringToToolTip(sDimNames);
-                int w = calculateMaxW(dims);
-                int x = GuiDraw.getMousePosition().x > Minecraft.getMinecraft().currentScreen.width / 2 ? this.getGuiWidth() - w - 8 : 0;
-                if (dims.size() > 10) {
-                    List<String> dims2 = dims.subList(11, dims.size());
-                    int w2 = calculateMaxW(dims2);
-                    dims = dims.subList(0, 11);
-                    w = calculateMaxW(dims);
-                    GuiDraw.drawMultilineTip(x == 0 ? 16 + w : x - (w2 + 8), 108 - (dims.size() * 8), dims2);
-                }
-                GuiDraw.drawMultilineTip(x, 108 - (dims.size() * 8), dims);
-
-                ttDisplayed = hideBackground;
+    /**
+     * Add lines to the current tooltip if appropriate
+     * 
+     * @param gui An instance of the currentscreen
+     * @param currenttip The current tooltip, will contain item name and info
+     * @param recipe The recipe index being handled
+     * @return The modified tooltip. DO NOT return null
+     */
+    @Override
+    public List<String> handleTooltip(GuiRecipe gui, List<String> currenttip, int recipe) {
+        if (toolTips && GuiContainerManager.shouldShowTooltip(gui) && currenttip.size() == 0) {
+            String dimNames = getDimensionNames(recipe);
+            Rectangle dimRect = getDimensionNamesRect(gui, recipe , dimNames);
+            Point mousePos = GuiDraw.getMousePosition();
+            
+
+            if (dimRect.contains(mousePos.x, mousePos.y)) {
+                List<String> dims = DimensionHelper.convertCondensedStringToToolTip(dimNames);
+                currenttip.addAll(dims);
             }
         }
+
+        return super.handleTooltip(gui, currenttip, recipe);
+    }
+
+    /**
+     * The dimension names for a given recipe index
+     * 
+     * @param The recipe index being handled
+     * @return A CSV string of dimension name abbreviations
+     */
+    protected abstract String getDimensionNames(int recipe);
+
+    /**
+     * Produce a rectangle covering the area of displayed dimension names
+     * 
+     * @param gui An instance of the currentscreen
+     * @param recipe The recipe index being handled
+     * @param dimNames Dimension names to produce a rectangle for
+     * @return Rectangle area of dimension names
+     */
+    protected Rectangle getDimensionNamesRect(GuiRecipe gui, int recipe, String dimNames) {
+        int height = dimNames.length() > 70 ? 30 : (dimNames.length() > 36 ? 20 : 10);        
+        Point offset = gui.getRecipePosition(recipe);
+        return new Rectangle(gui.guiLeft + offset.x + 2, gui.guiTop + offset.y + 110, gui.xSize - 9, height );
```

Comment:
This is the only place where Access Transformers/Reflection is currently of use. I've got a Reflection commit ready if we don't want to handle this differently.